internal/entitlement/connector.go

```diff
@@ -57,50 +57,74 @@ func NewEntitlementConnector(
 	}
 }
 
-func (c *entitlementConnector) CreateEntitlement(ctx context.Context, input CreateEntitlementInputs) (Entitlement, error) {
-	feature, err := c.featureConnector.GetFeature(ctx, models.NamespacedID{Namespace: input.Namespace, ID: input.FeatureID})
-	if err != nil {
-		return Entitlement{}, &productcatalog.FeatureNotFoundError{ID: input.FeatureID}
+func (c *entitlementConnector) CreateEntitlement(ctx context.Context, input CreateEntitlementInputs) (*Entitlement, error) {
+	// ID has precedence over key
+	idOrKey := input.FeatureID
```

Comment:
I think the inputs should already get a single FeatureIDorKey field, and this should be handled in the httpdriver layer.

This would also align the PR with the GetEntitlementValue's signature